rclpy/test/test_time_source.py

```diff
@@ -67,6 +68,19 @@ def publish_reversed_clock_messages(self):
             rclpy.spin_once(self.node, timeout_sec=1, executor=executor)
             time.sleep(1)
 
+    def set_use_sim_time_parameter(self, value):
+        self.node.set_parameters(
+            [Parameter('use_sim_time', Parameter.Type.BOOL, value)])
+        executor = rclpy.executors.SingleThreadedExecutor(context=self.context)
+
+        while rclpy.ok(context=self.context):
```

Comment:
In case of an error, it might be good to add a timeout here (e.g. 5 seconds) so that we don't wait indefinitely.

Reply:
Fixed